lib/web_ui/lib/src/ui/painting.dart

```diff
@@ -1845,3 +1849,109 @@ class ImageShader extends Shader {
         'ImageShader not implemented for web platform.');
   }
 }
+
+
+/// A handle to a read-only byte buffer that is managed by the engine.
+class ImmutableBuffer {
+  ImmutableBuffer._(this.length);
+
+  /// Creates a copy of the data from a [Uint8List] suitable for internal use
+  /// in the engine.
+  static Future<ImmutableBuffer> fromUint8List(Uint8List list) async {
+    final ImmutableBuffer instance = ImmutableBuffer._(list.length);
+    instance._list = list;
+    return instance;
+  }
+
+  Uint8List _list;
+
+  /// The length, in bytes, of the underlying data.
+  final int length;
+
+  /// Retreives a single byte from the underlying data.
+  int operator [](int index) => _list[index];
+
+  /// Retreives a single byte from the underlying data.
+  int elementAt(int index) => _list.elementAt(index);
+
+  /// Release the resources used by this object. The object is no longer usable
+  /// after this method is called.
+  void dispose() => _list = null;
+}
+
+/// A descriptor of data that can be turned into an [Image] via a [Codec].
+///
+/// Use this class to determine the height, width, and byte size of image data
+/// before decoding it.
+class ImageDescriptor {
+  ImageDescriptor._() : _width = null, _height = null, _rowBytes = null, _format = null;
+
+  /// Creates an image descriptor from encoded data in a supported format.
+  static Future<ImageDescriptor> encoded(ImmutableBuffer buffer) async {
+    final ImageDescriptor descriptor = ImageDescriptor._();
+    descriptor._data = buffer._list;
+    return descriptor;
+  }
+
+  /// Creates an image descriptor from raw image pixels.
+  ///
+  /// The `pixels` parameter is the pixel data in the encoding described by
+  /// `format`.
+  ///
+  /// The `rowBytes` parameter is the number of bytes consumed by each row of
+  /// pixels in the data buffer. If unspecified, it defaults to `width` multiplied
+  /// by the number of bytes per pixel in the provided `format`.
+  // Not async because there's no expensive work to do here.
+  ImageDescriptor.raw(
+    ImmutableBuffer buffer, {
+    required int width,
+    required int height,
+    int? rowBytes,
+    required PixelFormat pixelFormat,
+  }) : _width = width, _height = height, _rowBytes = rowBytes, _format = pixelFormat {
+    _data = buffer._list;
+  }
+
+  Uint8List _data;
+  final int? _width;
+  final int? _height;
+  final int? _rowBytes;
+  final PixelFormat? _format;
+
+  /// The width, in pixels, of the image.
+  int get width => _width != null ? _width : throw UnsupportedError('ImageDescriptor.width is not supported on web.');
```

Comment:
@jonahwilliams  - is there some way we can implement these? Should I just be filing a bug?

Reply:
There is no way these can be implemented on the html backend as far as I know. Fyi @yjbanov 

For the canvaskit backend, it might be feasible? @hterkelsen